swap/src/network.rs

```diff
@@ -17,3 +19,35 @@ impl Executor for TokioExecutor {
         let _ = self.handle.spawn(future);
     }
 }
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct Seed([u8; SEED_LENGTH]);
+
+impl Seed {
+    /// prefix "NETWORK" to the provided seed and apply sha256
+    pub fn new(seed: [u8; crate::seed::SEED_LENGTH]) -> Self {
+        let mut engine = sha256::HashEngine::default();
+
+        engine.input(&seed);
+        engine.input(b"NETWORK");
+
+        let hash = sha256::Hash::from_engine(engine);
+        Self(hash.into_inner())
+    }
+
+    pub fn bytes(&self) -> [u8; SEED_LENGTH] {
```

Comment:
```suggestion
    fn bytes(&self) -> [u8; SEED_LENGTH] {
```
This is not publicly used